pyproject.toml

```diff
@@ -3,6 +3,11 @@ requires = ["setuptools>=51.1.1", "wheel", "setuptools_scm[toml]>=5.0"]
 
 [tool.mypy]
 exclude = "build/"
+follow_imports = "normal"
+ignore_missing_imports = true
+warn_redundant_casts = true
+warn_unused_configs = true
+warn_unused_ignores = true
```

Comment:
See comment in the issue, but I don't want to activate any of these flags

Reply:
Are we aligned on the fact that those flags are actually improving type checks?

For instance, if mypy tells us that a `type: ignore` comment is unused (thanks to the `warn_unused_ignores` flag), then we should remove it because mypy doesn't take t into account at all. So it's just useless code.

I will close both PR, and issue, if you confirm you do not want such flags :)

Reply:
I'll admit not reading that closely, in in favor of warning more strictly, but I don't want it to ignore unused imports. The others looks fine, but it might help to have a comment about what they do above each

Reply:
I have had issues with `warn_unused_ignores` in the past when I tried enabling it.
The issue with `mypy` is that its results are not always reproducible. Meaning, in CI/CD, it may fail, but locally, it may be considered an "unused type ignore", which is frustrating. Because of this, I am in favor of occasionally running adhoc with the flag for `--warn-unused-ignores` and opening a PR that way.